libnmstate/nm/nmclient.py

```diff
@@ -220,3 +234,53 @@ def _register_first_action(self):
     def _execute_action_once(self, _):
         self.execute_next_action()
         return False
+
+
+class _NmClient:
+    def __init__(self):
+        self.client = NM.Client.new(None)
+        logging.debug("NM.Client created")
+        self._mainloop = None
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, _exc_type, _exc_value, _traceback):
+        pass
+
+    def __del__(self):
```

Comment:
Why do you use `__del__`  and not an explicit method (e.g. `delete_client`)?
`__del__` is called by the gc an an unknown time, which is dangerous and unpredictable.